### ISSUE TITLE
Cannot install Cython on Py3.10a1

### ISSUE THREAD

**Issue by neonene** [cython/cython#3876]
Python master after [037245c](https://github.com/python/cpython/tree/037245c5ac46c3436f617a1f5d965929754be239) fails to setup Cython with following error on Windows.
```
FusedNode.c(29697): warning C4013: '_PyGen_Send' undefined``
FusedNode.obj : error LNK2001: unresolved external symbol _PyGen_Send
```
Thanks.

**Comment by scoder** [comment]
Py 3.10 [now provides](https://bugs.python.org/issue41756) a new `PyIter_Send()` function replacing some code that Cython currently ships internally. It's not clear yet whether `_PyGen_Send()` will continue to be gone (and there hasn't been a release without it yet), but we should switch to the new function in Py3.10+ (and maybe use our own copy of it in older versions).

PR welcome. The C code for that is in `Coroutine.c`.

**Comment by neonene** [comment]
Could anyone please make a patch?

**Comment by scoder** [comment]
Let's wait for the back-and-forth in CPython's C-API to settle a bit. It's still very early in the release cycle of CPython 3.10.

I would at least like to wait for the results of the CPython core dev sprint, which is this week. The next alpha release will probably show where things are going. Afterwards, there's still plenty of time to follow up.

**Comment by scoder** [comment]
Resolved in b37607f46b165e81f7af2b4eec2ae7f76f7717df by @neonene.

**Comment by vstinner** [comment]
> Resolved in b37607f by @neonene.

I don't understand this change. It uses PyGen_Send() function on Python 3.10, but this function never existed in any Python version. Do you mean PyIter_Send()?

How did you test this commit? Was it pushed directly with no PR?

Should I reopen this issue or open a new issue?

**Comment by vstinner** [comment]
> I don't understand this change. It uses PyGen_Send() function on Python 3.10, but this function never existed in any Python version.

Hum. It seems like the PyGen_Send() function was removed in CPython by: https://github.com/python/cpython/commit/24a54c0bd48d9f6f1a1289ca57afb381bc4b280e

So maybe it existed for a few day in the master branch, but not in any release version. I'm now confused.

**Comment by scoder** [comment]
> I don't understand this change.

Ah, ok, sorry, didn't have much time do apply it, let alone test it with Py3.10, and there was a PR, so I just committed that. Yes, It should be `PyIter_Send()`, which was previously created from `PyGen_Send()` (which got removed afterwards).

Anyway, that commit was intended to be just a quick fix to get things working at all. It was not meant to be the final word. See https://github.com/cython/cython/pull/3879#issuecomment-713348218
